File: 49. Group Anagrams/GroupAnagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        p =[]
        d = {}
        for i in range(len(strs)):
            x = ""
            lst = [0] * 26
            for j in range(len(strs[i])):
                lst[ord(strs[i][j])-ord('a')]+=1
            for k in range(len(lst)):
                if(lst[k]!=0):
                    x+=chr(k+97)*lst[k];
            if(x not in d):
                d[x]=list()
            d[x].append(strs[i])
        l=[]
        for key,values in d.items():
            l.append(values)
        return(l)
            
     

File: 49. Group Anagrams/GroupAnagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) 
    {
        unordered_map<string,vector<string>> mymap;
        
       for(int i=0;i<strs.size();i++)
       {
          
           string sortedString = getSortedStr(strs[i]);
           mymap[sortedString].push_back(strs[i]); 
    
       }
       
        vector<vector<string>> ans;
    
        for (auto it : mymap)
        {
            
            ans.push_back(it.second);
        }
           
        return ans;
         
    }
    
    string getSortedStr(string s)
    {
        string str;
        vector<int>alpha(256, 0);
        
        for (int i = 0; i < s.size(); i++)
        {
            alpha[s[i]]++;            
        }
        
        
        for (int j = 0; j < 256; j++)
        {            
            if (alpha[j] > 0)
                str += string(alpha[j], (char)j); 
        }
        
        return str;
    }
};
